2023-05-03 16:19:14 Error: /paddle/paddle/phi/kernels/funcs/elementwise_functor.h:91 Assertion `b != 0` failed. InvalidArgumentError: Integer division by zero encountered in (floor) divide. Please check the input value.
2023-05-03 16:19:14 E0000 00:00:00.000000 20316 cuda_module.cc:110] RAW: PTX JIT ERROR LOG: 
2023-05-03 16:19:14 .
2023-05-03 16:19:14 F0000 00:00:00.000000 20316 cuda_module.cc:115] RAW: The error `CUDA_ERROR_LAUNCH_FAILED` occurs while compiling the ptx! And its message is `unspecified launch failure`.
2023-05-03 16:19:14 
2023-05-03 16:19:14 
2023-05-03 16:19:14 --------------------------------------
2023-05-03 16:19:14 C++ Traceback (most recent call last):
2023-05-03 16:19:14 --------------------------------------
2023-05-03 16:19:14 0   cinn::hlir::framework::GraphCompiler::Build(cinn::hlir::framework::GraphCompiler::CompileOptions const&, std::unordered_set<std::string, std::hash<std::string >, std::equal_to<std::string >, std::allocator<std::string > >&&, void*)
2023-05-03 16:19:14 1   cinn::hlir::framework::ParallelCompiler::operator()()
2023-05-03 16:19:14 2   cinn::hlir::framework::ParallelCompiler::LaunchTask()
2023-05-03 16:19:14 3   cinn::hlir::framework::RunTask(cinn::hlir::framework::ParallelCompiler::Task*)
2023-05-03 16:19:14 4   cinn::hlir::framework::ParallelCompiler::Task::CodegenAndJit()
2023-05-03 16:19:14 5   cinn::runtime::cuda::CUDAModule::GetFunction(int, std::string const&)
2023-05-03 16:19:14 6   google::RawLog__(int, char const*, int, char const*, ...)
2023-05-03 16:19:14 7   google::LogMessage::SendToSink()
2023-05-03 16:19:14 8   google::InstallFailureFunction(void (*)())
2023-05-03 16:19:14 
2023-05-03 16:19:14 ----------------------
2023-05-03 16:19:14 Error Message Summary:
2023-05-03 16:19:14 ----------------------
2023-05-03 16:19:14 FatalError: `Process abort signal` is detected by the operating system.
2023-05-03 16:19:14   [TimeInfo: *** Aborted at 1683101954 (unix time) try "date -d @1683101954" if you are using GNU date ***]
2023-05-03 16:19:14   [SignalInfo: *** SIGABRT (@0x4f5c) received by PID 20316 (TID 0x7f9084dc7740) from PID 20316 ***]
2023-05-03 16:19:14 
2023-05-03 16:19:14 Child aborted

看报错，应该是在paddle那边，这边CINN无法测试int类型。

--- a/python/tests/ops/test_divide_op.py
+++ b/python/tests/ops/test_divide_op.py
@@ -92,8 +92,98 @@ class TestDivAll(TestCaseHelper):
                 "y_shape": [1, 5, 2],
             },
             {
-                "x_shape": [2, 2, 32],
+                "x_shape": [16, 8, 4, 2],
+                "y_shape": [16, 8, 4, 2],
+            },
+            {
+                "x_shape": [16, 8, 4, 2, 1],
+                "y_shape": [16, 8, 4, 2, 1],
+            },
+        ]
+        self.dtypes = [
+            {
+                "x_dtype": "int32",
+                "y_dtype": "int32",
+            },
+            {
+                "x_dtype": "int64",
+                "y_dtype": "int64",
+            },
+            {
+                "x_dtype": "float32",
+                "y_dtype": "float32",
+            },
+            {
+                "x_dtype": "float64",
+                "y_dtype": "float64",
+            },
+        ]
+        self.attrs = []
+
+
+class TestDivNegOp(OpTest):
+    def setUp(self):
+        print(f"\nRunning {self.__class__.__name__}: {self.case}")
+        self.init_case()
+
+    def init_case(self):
+        self.x_np = self.random(
+            shape=self.case["x_shape"],
+            dtype=self.case["x_dtype"],
+            low=-10,
+            high=10)
+        self.y_np = self.random(
+            shape=self.case["y_shape"],
+            dtype=self.case["y_dtype"],
+            low=-10,
+            high=-1)
+
+    def build_paddle_program(self, target):
+        x = paddle.to_tensor(self.x_np, stop_gradient=True)
+        y = paddle.to_tensor(self.y_np, stop_gradient=True)
+
+        out = paddle.divide(x, y)
+
+        self.paddle_outputs = [out]
+
+    def build_cinn_program(self, target):
+        builder = NetBuilder("div")
+        x = builder.create_input(
+            self.nptype2cinntype(self.case["x_dtype"]), self.case["x_shape"],
+            "x")
+        y = builder.create_input(
+            self.nptype2cinntype(self.case["y_dtype"]), self.case["y_shape"],
+            "y")
+        out = builder.divide(x, y)
+
+        prog = builder.build()
+        res = self.get_cinn_output(prog, target, [x, y],
+                                   [self.x_np, self.y_np], [out])
+
+        self.cinn_outputs = [res[0]]
+
+    def test_check_results(self):
+        max_relative_error = self.case[
+            "max_relative_error"] if "max_relative_error" in self.case else 1e-5
+        self.check_outputs_and_grads(max_relative_error=max_relative_error)
+
+
+class TestDivNegAll(TestCaseHelper):
+    def init_attrs(self):
+        self.class_name = "TestDivNegOpCase"
+        self.cls = TestDivNegOp
+        self.inputs = [
+            {
+                "x_shape": [32],
                 "y_shape": [32],
+            },
+            {
+                "x_shape": [32, 64],
+                "y_shape": [32, 64],
+            },
+            {
+                "x_shape": [2, 3, 4],
+                "y_shape": [1, 5, 2],
             },
             {
                 "x_shape": [16, 8, 4, 2],
@@ -127,3 +217,4 @@ class TestDivAll(TestCaseHelper):
 
 if __name__ == "__main__":
     TestDivAll().run()
+    TestDivNegAll().run()

--- a/python/tests/ops/test_divide_op.py
+++ b/python/tests/ops/test_divide_op.py
@@ -41,7 +41,7 @@ class TestDivOp(OpTest):
         self.y_np = self.random(
             shape=self.case["y_shape"],
             dtype=self.case["y_dtype"],
-            low=-10,
+            low=1,
             high=10)
 
     def build_paddle_program(self, target):
@@ -80,31 +80,39 @@ class TestDivAll(TestCaseHelper):
         self.cls = TestDivOp
         self.inputs = [
             {
-                "x_shape": [2, 3, 4],
-                "y_shape": [1, 5, 2],
+                "x_shape": [32],
+                "y_shape": [32],
             },
             {
                 "x_shape": [32, 64],
                 "y_shape": [32, 64],
             },
             {
+                "x_shape": [2, 3, 4],
+                "y_shape": [1, 5, 2],
+            },
+            {
                 "x_shape": [2, 2, 32],
                 "y_shape": [32],
             },
             {
-                "x_shape": [32],
-                "y_shape": [32],
+                "x_shape": [16, 8, 4, 2],
+                "y_shape": [16, 8, 4, 2],
+            },
+            {
+                "x_shape": [16, 8, 4, 2, 1],
+                "y_shape": [16, 8, 4, 2, 1],
             },
         ]
         self.dtypes = [
-            #{
-            #    "x_dtype": "int32",
-            #    "y_dtype": "int32",
-            #},
-            #{
-            #    "x_dtype": "int64",
-            #    "y_dtype": "int64",
-            #},
+            {
+                "x_dtype": "int32",
+                "y_dtype": "int32",
+            },
+            {
+                "x_dtype": "int64",
+                "y_dtype": "int64",
+            },
             {
                 "x_dtype": "float32",
                 "y_dtype": "float32",

--- a/python/tests/ops/test_divide_op.py
+++ b/python/tests/ops/test_divide_op.py
@@ -97,14 +97,14 @@ class TestDivAll(TestCaseHelper):
             },
         ]
         self.dtypes = [
-            {
-                "x_dtype": "int32",
-                "y_dtype": "int32",
-            },
-            {
-                "x_dtype": "int64",
-                "y_dtype": "int64",
-            },
+            #{
+            #    "x_dtype": "int32",
+            #    "y_dtype": "int32",
+            #},
+            #{
+            #    "x_dtype": "int64",
+            #    "y_dtype": "int64",
+            #},
             {
                 "x_dtype": "float32",
                 "y_dtype": "float32",

